test/unit/errors.test.js

```diff
@@ -2,10 +2,61 @@
 
 const expect = require('chai').expect;
 const { getSymbolFrom } = require('../tools/utils');
-const MongoNetworkError = require('../../src/error').MongoNetworkError;
+const {
+  MongoError,
+  MongoServerError,
+  MongoDriverError,
+  MongoNetworkError,
+  MongoNetworkTimeoutError,
+  MongoParseError,
+  MongoSystemError,
+  MongoServerSelectionError,
+  MongoWriteConcernError
+} = require('../../src/error');
+const { MongoBulkWriteError } = require('../../src/bulk/common');
+const { PoolClosedError, WaitQueueTimeoutError } = require('../../src/cmap/errors');
+
+describe('MongoErrors', () => {
+  const errorClasses = {
+    MongoError,
+    MongoServerError,
+    MongoDriverError,
+    MongoNetworkTimeoutError,
+    MongoParseError,
+    MongoSystemError,
+    MongoServerSelectionError,
+    MongoWriteConcernError,
+    MongoBulkWriteError,
+    PoolClosedError,
+    WaitQueueTimeoutError
+  };
+
+  for (const errorName in errorClasses) {
+    describe(errorName, () => {
+      it(`name should be read-only`, () => {
+        // Dynamically create error class with message
+        let error = new errorClasses[errorName]('generated by test');
+
+        // expect name property to be class name
+        expect(error.name).to.eql(errorName);
+        expect(() => {
+          error.name = 'renamed by test';
+        }).to.throw(TypeError);
```

Comment:
I have a suggestion that results in essentially equivalent assertion but checks precisely what we desire from all of our errors:

```javascript
const errorPrototype = Object.getPrototypeOf(error)
expect(errorPrototype).to.have.ownPropertyDescriptor('name').that.has.property('get')
```